socketio/src/client/socket.rs

```diff
@@ -192,13 +214,29 @@ impl SocketBuilder {
     /// assert!(result.is_ok());
     /// ```
     pub fn connect(self) -> Result<Socket> {
-        //TODO: Add check for empty path and set to /socket.io/
-        let mut socket = Socket::new(
-            self.address,
-            self.namespace,
-            self.tls_config,
-            self.opening_headers,
-        )?;
+        let mut url = Url::parse(&self.address)?;
+
+        if url.path() == "/" {
+            url.set_path("/socket.io/");
+        }
+
+        let mut builder = EngineIoSocketBuilder::new(url);
+
+        if let Some(tls_config) = self.tls_config {
+            builder = builder.tls_config(tls_config);
+        }
+        if let Some(headers) = self.opening_headers {
+            builder = builder.headers(headers);
+        }
```

Comment:
Would it make sense to have `builder.headers()` or `builder.tls_config()` taking an `Option<T>`? That way we wouldn't have the if let here and the builder holds them as `Option` anyways.

Reply:
But it feels strange to call a setter with an optional value...

Reply:
> But it feels strange to call a setter with an optional value...

Ultimately our primary goal is to provide an easy to use interface to the consumers of this API, if that means we need to have a tiny bit more boilerplate to save the users from writing Some() around every call to the setters headers and tls_config so be it. In the future this EngineIO builder will also be public.

Reply:
Yep that's true.